pkg/version/channels.go

```diff
@@ -59,20 +58,17 @@ func (c Channels) Match(actualVersion string) error {
 
 // GetLatestVersions performs an online request to check for the latest Linkerd
 // release channels.
-func GetLatestVersions(uuid string, source string) (Channels, error) {
+func GetLatestVersions(ctx context.Context, uuid string, source string) (Channels, error) {
 	url := fmt.Sprintf(versionCheckURL, Version, uuid, source)
-	return getLatestVersions(http.DefaultClient, url, uuid, source)
+	return getLatestVersions(ctx, http.DefaultClient, url, uuid, source)
 }
 
-func getLatestVersions(client *http.Client, url string, uuid string, source string) (Channels, error) {
+func getLatestVersions(ctx context.Context, client *http.Client, url string, uuid string, source string) (Channels, error) {
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		return Channels{}, err
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
just confirming, that by delegating context creation to the healthcheck caller, we're increasing this timeout from 5s to 30s. i think this is ok?

Reply:
That's correct. To cater for the case of slow clusters like described in #2011